src/apis/impl/DataSampleApiImpl.ts

```diff
@@ -863,6 +865,35 @@ export class DataSampleApiImpl implements DataSampleApi {
       })
   }
 
+  async giveAccessToMany(dataSamples: Array<DataSample>, delegatedTo: string): Promise<Array<string>> {
+    return (await Promise.all(dataSamples.map(async dataSample => {
+      try {
+        return (await this.giveAccessTo(dataSample, delegatedTo)).id
```

Comment:
This is an operation that's gonna take a lot of time which implies two things to do : 
- Try to avoid any additional requests. I would for example save all contacts at the same time, instead of treating them one by one. 
- Add a service `mayAccess(dataSample, dataOwnerId)`, so that if a problem happens during the treatment (app shutdown, ...) the user will be able to continue its treatment easily, taking back the ones that haven't been treated. 
We need to think about this point together, let's discuss it